### PR TITLE
Leading space in to_string, #11833

### LINKED ISSUE #11833: Leading space in to_string(index=False)
#11750
```python 
In [2]: df = pd.DataFrame({'a':range(5)})

In [3]: df.to_string()
Out[3]: '   a\n0  0\n1  1\n2  2\n3  3\n4  4'

In [4]: df.to_string(index=False)
Out[4]: ' a\n 0\n 1\n 2\n 3\n 4'
```
What is the use of a leading space when index = False ?

### PR DESCRIPTION
**Pull request by Vutsuak16**
closes #11833 

Final Commits

### CODE REVIEW

**Comment by Vutsuak16** [comment]
@jreback  @jorisvandenbossche  it has passed the test time !!
Had to link with travis to run them locally
Please have a look at the commit

**Comment by jreback** [comment]
needs a note in whatsnew, with an example as this is techincally an API change.

File: pandas/core/format.py [hunk 1]
```diff
@@ -578,7 +578,11 @@ def to_string(self):
                 strcols = self._to_str_columns()
                 text = self.adj.adjoin(1, *strcols)
 
-        self.buf.writelines(text)
+        if self.index==True:
+            self.buf.writelines(text)
```
Comment (jreback) [inline pandas/core/format.py]: more like:

```
# put a nice comment here
if not self.index:
     text = text.replace(......)
self.buf.writelines(text)
```

File: pandas/tests/test_format.py [hunk 2]
```diff
@@ -1725,7 +1725,7 @@ def test_to_string_no_index(self):
                         'y': [4, 5, 6]})
 
         df_s = df.to_string(index=False)
-        expected = " x  y\n 1  4\n 2  5\n 3  6"
+        expected = "x  y\n1  4\n2  5\n3  6"
```
Comment (jreback) [inline pandas/tests/test_format.py]: since we don't appear to have a series tests for this, pls add one

**Comment by Vutsuak16** [comment]
@jreback   @jorisvandenbossche  do i have to update the what's new section in the documentation?
I have no experience with Pandas workflow, please guide me here?

**Comment by TomAugspurger** [comment]
http://pandas.pydata.org/pandas-docs/stable/contributing.html, specifically http://pandas.pydata.org/pandas-docs/stable/contributing.html#documenting-your-code for the whatsnew.

**Comment by Vutsuak16** [comment]
@TomAugspurger  @jreback 
I did the changes in the documentation
With the recent commit

File: doc/source/whatsnew/v0.18.0.txt [hunk 3]
```diff
@@ -161,6 +161,46 @@ In addition, ``.round()`` will be available thru the ``.dt`` accessor of ``Serie
...
+
+- the "leading whitespaces" have been removed from the output of "to_string(index=False)" method (:issue: '11833')
+
+Previous Behaviour for DataFrame Objects
+
+.. ipython:: python
+
+   df=pd.DataFrame({'a':range(5),'b':range(5)})
+   df.to_string(index=False)
+   u' a  b\n 0  0\n 1  1\n 2  2\n 3  3\n 4  4'
```
Comment (jreback) [inline doc/source/whatsnew/v0.18.0.txt]: you need to mark this ``code-block:: python``, then paste in actual ipython output

File: doc/source/whatsnew/v0.18.0.txt [hunk 4]
```diff
@@ -161,6 +161,46 @@ In addition, ``.round()`` will be available thru the ``.dt`` accessor of ``Serie
...
+
+
+New Behaviour for DataFrame Objects
+
+.. ipython:: python
+
+   df=pd.DataFrame({'a':range(5),'b':range(5)})
+   df.to_string(index=False)
+   u'a  b\n0  0\n1  1\n2  2\n3  3\n4  4'
```
Comment (jreback) [inline doc/source/whatsnew/v0.18.0.txt]: you don't need the output line, ``ipython`` will make it here

File: doc/source/whatsnew/v0.18.0.txt [hunk 5]
```diff
@@ -161,6 +161,46 @@ In addition, ``.round()`` will be available thru the ``.dt`` accessor of ``Serie
...
+Change in to_string(index=False) method of DataFrameFormatter and SeriesFormatter class
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- the "leading whitespaces" have been removed from the output of "to_string(index=False)" method (:issue: '11833')
+
+Previous Behaviour for DataFrame Objects
+
+.. ipython:: python
+
+   df=pd.DataFrame({'a':range(5),'b':range(5)})
```
Comment (jreback) [inline doc/source/whatsnew/v0.18.0.txt]: put the first line in its own ``ipython:: python`` block

File: doc/source/whatsnew/v0.18.0.txt [hunk 6]
```diff
@@ -161,6 +161,46 @@ In addition, ``.round()`` will be available thru the ``.dt`` accessor of ``Serie
...
+New Behaviour for DataFrame Objects
+
+.. ipython:: python
+
+   df=pd.DataFrame({'a':range(5),'b':range(5)})
+   df.to_string(index=False)
+   u'a  b\n0  0\n1  1\n2  2\n3  3\n4  4'
+
+
+Previous Behaviour for Series Objects
```
Comment (jreback) [inline doc/source/whatsnew/v0.18.0.txt]: skip doing this for Series (just say it applies both to Series & DataFrame above)

File: doc/source/whatsnew/v0.18.0.txt [hunk 7]
```diff
@@ -161,6 +161,46 @@ In addition, ``.round()`` will be available thru the ``.dt`` accessor of ``Serie
 
 .. _whatsnew_0180.api_breaking:
 
+
+Change in to_string(index=False) method of DataFrameFormatter and SeriesFormatter class
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (jreback) [inline doc/source/whatsnew/v0.18.0.txt]: the underline must match the length of the stop. be a bit more concise on the title.

File: doc/source/whatsnew/v0.18.0.txt [hunk 8]
```diff
@@ -161,6 +161,46 @@ In addition, ``.round()`` will be available thru the ``.dt`` accessor of ``Serie
...
+
+- the "leading whitespaces" have been removed from the output of "to_string(index=False)" method (:issue: '11833')
```
Comment (jreback) [inline doc/source/whatsnew/v0.18.0.txt]: no space after ``:issue:``

File: pandas/core/format.py [hunk 9]
```diff
@@ -578,6 +588,18 @@ def to_string(self):
                 strcols = self._to_str_columns()
                 text = self.adj.adjoin(1, *strcols)
 
+        ''' this part checks for the argument index passed.
+         If it is true then  returns the text with index mentioned else it returns without the index,
+          the leading whitespaces ,in place of index are also omitted when the index is False
+          ex:
```
Comment (jreback) [inline pandas/core/format.py]: only need a single line comment (what I meant)

File: pandas/core/format.py [hunk 10]
```diff
@@ -242,10 +242,20 @@ def to_string(self):
             fmt_values.insert(row_num + n_header_rows, dot_str)
             fmt_index.insert(row_num + 1, '')
 
+        ''' this part checks for the argument index passed.
```
Comment (jreback) [inline pandas/core/format.py]: only need a single line comment

**Comment by jreback** [comment]
- some comments
- remove the html-styling file from the commits.
- squash into a single commit.

**Comment by Vutsuak16** [comment]
@jreback  is there anything else that needs attention , for the issue #11833 , I did the docs changes as directed by you

File: pandas/core/format.py [hunk 11]
```diff
@@ -578,6 +580,11 @@ def to_string(self):
                 strcols = self._to_str_columns()
                 text = self.adj.adjoin(1, *strcols)
 
+
+        '''condition check for index '''
```
Comment (jreback) [inline pandas/core/format.py]: here too

File: pandas/core/format.py [hunk 12]
```diff
@@ -242,10 +242,12 @@ def to_string(self):
             fmt_values.insert(row_num + n_header_rows, dot_str)
             fmt_index.insert(row_num + 1, '')
 
+        ''' condition check for index '''
```
Comment (jreback) [inline pandas/core/format.py]: take out these comments

File: doc/source/whatsnew/v0.18.0.txt [hunk 13]
```diff
@@ -161,6 +161,30 @@ In addition, ``.round()`` will be available thru the ``.dt`` accessor of ``Serie
 
 .. _whatsnew_0180.api_breaking:
 
+
+Changed to_string(index=False) method of DataFrame and Series formatter class.
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- the "leading whitespaces" have been removed from the output of "to_string(index=False)" method (:issue:'11833')
```
Comment (jreback) [inline doc/source/whatsnew/v0.18.0.txt]: on 2nd thought, let's just use this single line 168 instead, pls remove the section. also put double backticks around ``to_string(index=False)``

**Comment by jreback** [comment]
need to remove the html-styling ipynb that is included
pls squash to a single commit

**Comment by Vutsuak16** [comment]
@jreback  I did the needfull ,ignore the last commit :)

File: doc/source/whatsnew/v0.18.0.txt [hunk 14]
```diff
@@ -161,6 +161,35 @@ In addition, ``.round()`` will be available thru the ``.dt`` accessor of ``Serie
 
 .. _whatsnew_0180.api_breaking:
 
+
+<<<<<<< HEAD
```
Comment (jreback) [inline doc/source/whatsnew/v0.18.0.txt]: you have unresolved commits here

**Comment by jreback** [comment]
you need to squash: http://pandas.pydata.org/pandas-docs/stable/contributing.html#combining-commits

**Pull request closed without merging**